src/Features/Core/Portable/Diagnostics/Analyzers/QualifyMemberAccessDiagnosticAnalyzerBase.cs

```diff
@@ -90,27 +73,18 @@ private void AnalyzeOperation(OperationAnalysisContext context)
             var isQualificationPresent = IsAlreadyQualifiedMemberAccess(memberReference.Instance.Syntax);
             if (shouldOptionBePresent && !isQualificationPresent)
             {
-                DiagnosticDescriptor descriptor;
-                switch (optionValue.Notification.Value)
+                var severity = optionValue.Notification.Value;
+                if (severity != DiagnosticSeverity.Hidden)
                 {
-                    case DiagnosticSeverity.Hidden:
-                        descriptor = null;
-                        break;
-                    case DiagnosticSeverity.Info:
-                        descriptor = s_descriptorQualifyMemberAccessInfo;
-                        break;
-                    case DiagnosticSeverity.Warning:
-                        descriptor = s_descriptorQualifyMemberAccessWarning;
-                        break;
-                    case DiagnosticSeverity.Error:
-                        descriptor = s_descriptorQualifyMemberAccessError;
-                        break;
-                    default:
-                        throw ExceptionUtilities.Unreachable;
-                }
+                    var descriptor = new DiagnosticDescriptor(
```

Comment:
You can avoid instantiating a new descriptor by writing a helper extension method: `Diagnostic Create(this DiagnosticDescriptor descriptor, DiagnosticSeverity severity, Location location)` that invokes [Diagnostic.Create](http://source.roslyn.io/#Microsoft.CodeAnalysis/Diagnostic/Diagnostic.cs,153) with required fields and then report this diagnostic below.